src/utils/post_utils.js

```diff
@@ -175,3 +183,82 @@ export function comparePosts(a, b) {
 
     return 0;
 }
+
+export function combineSystemPosts(postsIds = [], posts = {}, channelId) {
+    if (postsIds.length === 0) {
+        return {postsForChannel: postsIds, nextPosts: posts};
+    }
+
+    const postsForChannel = [];
+    const nextPosts = {...posts};
+
+    let userActivitySystemPosts = [];
+    let systemPostIds = [];
+    let createAt;
+    let combinedPostId;
+
+    postsIds.forEach((p, i) => {
+        const channelPost = posts[p];
+        if (
+            channelPost.delete_at === 0 &&
+            (isUserActivityPost(channelPost.type) || channelPost.type === Posts.POST_TYPES.COMBINED_USER_ACTIVITY)
+        ) {
+            if (!createAt || createAt > channelPost.create_at) {
+                createAt = channelPost.create_at;
+            }
+
+            if (isUserActivityPost(channelPost.type)) {
+                userActivitySystemPosts = [...userActivitySystemPosts, channelPost];
```

Comment:
why are you creating a new array every time instead of just adding the post to the existing array?

Reply:
It's only out of convenience during my first iteration :) and just missed to clean up. Will update using push instead.